hubblestack/extmods/grains/docker_details.py

```diff
@@ -0,0 +1,76 @@
+"""
+HubbleStack Docker Details Grain.
+CLI Usage - hubble grains.get docker_details
+Example Output - {u'installed': True, u'running': True}
+Author - Mudit Agarwal (muagarwa@adobe.com)
+"""
+import salt.utils.platform
+import logging
+from hubblestack.utils.osquery_lib import query as osquery_util
+log = logging.getLogger(__name__)
+osquery_path = '/opt/osquery/osqueryi'
+
+def get_docker_details(grains):
+  try:
+    docker_grains = {}
+
+    if salt.utils.platform.is_windows():
+      log.debug('This grain is only available on linux')
+      return docker_grains
+
+    docker_details = {}
+    docker_details['installed'] = _is_docker_installed(grains)
+    docker_details['running'] = False
+
+    if docker_details['installed']:
+      docker_details['running'] = _is_docker_process_running()
```

Comment:
Instead of using osquery, which adds overhead, let's just use salt's built in package/process checker.
```
docker_details['installed'] = bool(__salt__['pkg.version']('docker'))
docker_details['running'] = bool(__salt__['ps.pgrep']('docker'))
```

Reply:
I thought of using it, but salt is initialised after the grains are loaded. And so the salt module is unavailable in the grains.

Reply:
Ah, got it. That makes sense.